sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala

```diff
@@ -493,8 +493,9 @@ private[spark] object HiveUtils extends Logging {
       val hiveTable = HiveClientImpl.toHiveTable(table)
       // Note: Hive separates partition columns and the schema, but for us the
       // partition columns are part of the schema
-      val partCols = hiveTable.getPartCols.asScala.map(HiveClientImpl.fromHiveColumn)
-      val dataCols = hiveTable.getCols.asScala.map(HiveClientImpl.fromHiveColumn)
+      val partCols = hiveTable.getPartCols.asScala.map(c =>
```

Comment:
Can you actually just catch the exception here and rethrow with improved error message?

Reply:
ok. i changed to rethrow exception.